kernel/kernel-default/src/main/java/io/delta/kernel/data/vector/AbstractColumnVector.java

```diff
@@ -0,0 +1,171 @@
+/*
+ * Copyright (2023) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.delta.kernel.data.vector;
+
+import static io.delta.kernel.DefaultKernelUtils.checkArgument;
+import java.util.List;
+import java.util.Map;
+import static java.util.Objects.requireNonNull;
+import java.util.Optional;
+
+import io.delta.kernel.data.ColumnVector;
+import io.delta.kernel.data.Row;
+import io.delta.kernel.types.DataType;
+
+/**
+ * Abstract implementation of {@link ColumnVector} that provides the default functionality
+ * common to most of the specific data type {@link ColumnVector} implementations.
+ */
+public abstract class AbstractColumnVector
+    implements ColumnVector
+{
+    private final int size;
+    private final DataType dataType;
+    private final Optional<boolean[]> nullability;
+
+    protected AbstractColumnVector(int size, DataType dataType, Optional<boolean[]> nullability)
+    {
+        checkArgument(size >= 0, "invalid size: %s", size);
+        this.size = size;
+        this.dataType = requireNonNull(dataType);
+        this.nullability = requireNonNull(nullability);
+    }
+
+    @Override
+    public DataType getDataType()
+    {
+        return dataType;
+    }
+
+    @Override
+    public int getSize()
+    {
+        return size;
+    }
+
+    @Override
+    public void close()
+    {
+        // By default, nothing to close, if the implementation has any resources to release
+        // it can override it
+    }
+
+    /**
+     * Is the value at given {@code rowId} index is null?
+     *
+     * @param rowId
+     * @return
+     */
+    @Override
+    public boolean isNullAt(int rowId)
```

Comment:
To clarify, I'm mainly wondering what's the expected behavior when no `nullability` array is provided? The default being all values are null in that scenario seems a little weird.

If we expect any child class that omits the `nullability` array to override this can we throw an error here when it isn't present?

Reply:
see the prev reply.